### ISSUE TITLE
Chapter 1: fermat's little theorem 

### ISSUE THREAD

**Issue by lyhistory** [jimmysong/programmingbitcoin#113]
quote from Page 14 "In case you didn't get it, the answer is that n^(p-1) is always 1 for every p that is prime and every n>0."

not every n>0, n has to coprime to p, right? 
and here is my test:
![image](https://user-images.githubusercontent.com/1522513/54624023-8f8eb900-4aa7-11e9-90e0-05a3ef6f84a7.png)
refer to https://en.wikipedia.org/wiki/Fermat%27s_little_theorem

**Comment by AtAFork** [comment]
Possible that I'm misunderstanding the text or your question, but I thought that the finite field doesn't include `p` itself (e.g. in your case of `range(1,prime+1)` where `p = 7`, the `list` produced by that Python `range` would be `[1,2,3,4,5,6,7]`, but per below Ch 1 screenshot it should be `[1,2,3,4,5,6]`, so your code should instead be:

etc etc etc ... `for x in range(1,prime)`


![image](https://user-images.githubusercontent.com/9468623/54778697-65421600-4be3-11e9-9b85-e4c71f6e8faa.png)

**Comment by lyhistory** [comment]
hi,  you're right about finite field, but I'm talking about n^(p-1), sorry for the misleading of the screenshot sample code, let me rephrase again:

basically what I mean is the author states that "n^(p-1) is always 1 for every p that is prime and every n>0."
first of all, not true when n not coprime with p "14^(7-1) MOD 7 = 0" ,
maybe I think here it implies that n lies in F(p), but why it emphasize n>0 instead? 
and aslo n not necessarily lies in finite field, as long as it coprime with p, the statement holds, so example "15^(7-1) MOD 7 = 1"

**Comment by jimmysong** [comment]
p being prime necessarily means n is coprime to p, no?
 
Please let me know if I'm not understanding something @lyhistory
